enhancement(config api): Add `graphql` field to toggle graphql endpoint (#19645)
vectordotdev/vector: changelog.d/graphql_endpoint_toggle.enhancement.md

> Added boolean `graphql` field to the api configuration to toggle graphql endpoint on/off in Vector.

Reviewer: nit: The "in Vector" part is redundant, as that is the overall context of all such changes.
```suggestion
Added a boolean `graphql` field to the api configuration to allow disabling the graphql endpoint.
```

> Note that the `playground` endpoint will now only be enabled if the `graphql` endpoint is enabled.

Reviewer: ```suggestion
Note that the `playground` endpoint will now only be enabled if the `graphql` endpoint is also enabled.
```

> Note that the `playground` endpoint will now only be enabled if the `graphql` endpoint is also enabled.

Reviewer 2: <!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏I'm not super familiar with this area but I'm wondering if this would be considered a breaking change? It it is that would probably be better to separate out into a second fragment addition.

Reviewer: I don't think it's breaking as the `playground` field defaults to true, as it was before.

Author: ^

Reviewer 2: roger that 👍
